web_src/js/features/notification.js

```diff
@@ -19,21 +19,53 @@ export function initNotificationsTable() {
 }
 
 export function initNotificationCount() {
-  if (NotificationSettings.MinTimeout <= 0) {
+  const notificationCount = $('.notification_count');
+
+  if (notificationCount.length <= 0) {
     return;
   }
 
-  const notificationCount = $('.notification_count');
+  if (NotificationSettings.EventSourceUpdateTime > 0 && !!window.EventSource) {
+    // Try to connect to the event source first
+    const source = new EventSource(`${AppSubUrl}/user/events`);
+    source.addEventListener('notification-count', async (e) => {
+      if (e.origin !== window.location.origin) {
+        return;
+      }
 
-  if (notificationCount.length > 0) {
-    const fn = (timeout, lastCount) => {
-      setTimeout(async () => {
-        await updateNotificationCountWithCallback(fn, timeout, lastCount);
-      }, timeout);
-    };
+      const data = JSON.parse(e.data);
```

Comment:
Is try/catch required? (e.g. invalid JSON)

Reply:
It's good practice to always wrap `JSON.parse` in try/catch.

Reply:
But we are the generators of this data - we should be guaranteeing that it's JSON...

Reply:
What do we do with the error when we catch it?  I've stuck a:

```js
try {
  ...
} catch(error) {
  console.error(error);
}
```

But I'm not certain that's much better than letting it propogate.

Reply:
I'd just `return`, no need to log it imho.

Reply:
I'll leave the console log in so at least a developer can figure out if something is wrong.